lib/appsignal/absinthe.ex

```diff
@@ -0,0 +1,54 @@
+defmodule Appsignal.Absinthe do
+  require Appsignal.Utils
+
+  @tracer Appsignal.Utils.compile_env(:appsignal, :appsignal_tracer, Appsignal.Tracer)
+  @span Appsignal.Utils.compile_env(:appsignal, :appsignal_span, Appsignal.Span)
+
+  @moduledoc false
+
+  require Logger
+
+  def attach do
+    handlers = %{
+      [:absinthe, :execute, :operation, :start] => &__MODULE__.absinthe_execute_operation_start/4,
+      [:absinthe, :execute, :operation, :stop] => &__MODULE__.absinthe_execute_operation_stop/4
+    }
+
+    for {event, fun} <- handlers do
+      detach = :telemetry.detach({__MODULE__, event})
+      attach = :telemetry.attach({__MODULE__, event}, event, fun, :ok)
+
+      case {detach, attach} do
+        {:ok, :ok} ->
+          _ =
+            Appsignal.IntegrationLogger.debug(
+              "Appsignal.Absinthe reattached to #{inspect(event)}"
+            )
+
+          :ok
+
+        {{:error, :not_found}, :ok} ->
+          _ =
+            Appsignal.IntegrationLogger.debug("Appsignal.Absinthe attached to #{inspect(event)}")
+
+          :ok
+
+        {_, {:error, _} = error} ->
+          Logger.warn("Appsignal.Absinthe not attached to #{inspect(event)}: #{inspect(error)}")
+
+          error
+      end
+    end
+  end
+
+  def absinthe_execute_operation_start(_event, _measurements, _metadata, _config) do
+    "http_request"
```

Comment:
```suggestion
    "graphql"
```
Let's set the namespace to "graphql".

Reply:
Interesting question. We can't simply set the namespace here, as this Span might be nested in a root span created by the Phoenix integration (https://github.com/appsignal/appsignal-elixir-phoenix/pull/67). What we could do is set the namespace of the root span (if it exists) by calling out to `Appsignal.Span.set_namespace/2`. I'll do some testing with that.

Reply:
I think it's also okay to set it only for this span, in case this span is the root span. Maybe it's kind of unexpected behaviour for regular web endpoints to be moved to the `graphql` namespace if they perform a GraphQL query?

Reply:
That's why the implementation is how it is right now. It makes sense to have this as an HTTP request, as that's essentially what it is. The user can of course always choose to namespace these through calling `Appsignal.Span.set_namespace/2` themselves.

I think moving this to a separate namespace only when it's not wrapped would be confusing, though.

Reply:
Hm, yeah, I can see how that can be confusing as well. That's what the Node.js one does, though. If it's a GraphQL query happening in the context of a web request, then the action is a web request and has the web namespace. If it's happening in its own, then the action is a GraphQL query and it goes in the GraphQL namespace.

That said I don't have a strong belief that the Node.js one does "the right thing" 😅 so feel free to pick whichever behaviour makes sense to you. Maybe we could even ship it without a custom namespace for now.